provider/pkg/schema/gen_tags.go

```diff
@@ -56,40 +62,73 @@ func (ctx *context) GetTagsStyle(propName string, typeSpec *pschema.TypeSpec) Ta
 		return TagsStyleStringMap
 	}
 
-	if ctx.tagStyleIsKeyValueArray(propName, typeSpec, false /* includeCreateOnly */) {
+	if ctx.tagStyleIsKeyValueArray(propName, typeSpec, false /* includeExtraProperties */, false /* allowOneOf */) {
```

Comment:
Calling this method three times with different combinations of boolean parameters is a bit complicated. Since the boolean params are only used to distinguish different kinds of tag arrays, could the method just return the right kind?
```go
tagArrayKind, isTagArray := ctx.tagStyleIsKeyValueArray(propName, typeSpec)
if isTagArray {
    return tagArrayKind
}
```

Reply:
Yup, good thought. Will refactor a little further 👍